app/javascript/components/search/ReviewedFilter.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from 'react-bootstrap/Button';
+import Popover from 'react-bootstrap/Popover';
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
+import Form from 'react-bootstrap/Form';
+import { inject, observer } from 'mobx-react';
+
+import { getButtonHintString } from './utils';
+
+@inject('filtersStore')
+@observer
+class FormFilter extends React.Component {
+  static propTypes = {
+    filtersStore: PropTypes.object,
+    onSubmit: PropTypes.func.isRequired,
+  };
+
+  handleClearReviewed = () => {
+    const { filtersStore, onSubmit } = this.props;
+
+    filtersStore.isReviewed = null;
+    onSubmit();
+  }
+
+  renderPopover = () => {
+    const { filtersStore, onSubmit } = this.props;
+    const { isReviewed, handleChangeIsReviewed } = filtersStore;
+
+    return (
+      <Popover
+        className="filters-popover"
+        id="reviewed-filter"
+      >
+        {/* Note: `id` is required for checkbox label to be clickable. */}
+        <Form.Check
+          id="is-reviewed"
+          type="checkbox"
+          label={I18n.t('filter.is_reviewed')}
+          checked={isReviewed || ''}
+          onChange={handleChangeIsReviewed}
+        />
+
+        <div className="btn-apply-container">
```

Comment:
I am still kind of uneasy about these buttons being duplicated on each one. It's like 15 lines of duplicated code. Could we not just make a subclass of Popover called like FilterPopover that contains this?

Reply:
The inner components for each popover are different enough that it's hard to unify the entire thing, but I'll at least make each individual component more generic. e.g.

```jsx
// QuestionFilter
<Popover
  className="filters-popover wide display-logic-container"
  id="form-filter"
>
  <ConditionSetFormField />

  <div className="btn-apply-container condition-margin">
    <Button
      className="btn-apply"
      onClick={onSubmit}
    >
      {I18n.t('common.apply')}
    </Button>
  </div>
</Popover>
```

vs.

```jsx
// ReviewedFilter
<Popover
  className="filters-popover"
  id="reviewed-filter"
>
  {/* Note: `id` is required for checkbox label to be clickable. */}
  <Form.Check
    id="is-reviewed"
    type="checkbox"
    label={I18n.t('filter.is_reviewed')}
    checked={isReviewed || ''}
    onChange={handleChangeIsReviewed}
  />

  <div className="btn-apply-container">
    <Button
      className="btn-clear btn-secondary"
      onClick={this.handleClearReviewed}
    >
      {I18n.t('common.clear')}
    </Button>
    <Button
      className="btn-apply"
      onClick={onSubmit}
    >
      {I18n.t('common.apply')}
    </Button>
  </div>
</Popover>
```

I can add common `FilterPopover`, `FilterPopoverButtonsContainer`, `ApplyButton`, `ClearButton`, etc. which will help.